rust/builder/mod.rs

```diff
@@ -83,46 +120,70 @@ pub fn rule(name: &str) -> RuleDeclaration {
-    UnboundVariable::hidden().type_(name.into())
+    UnboundConceptVariable::hidden().type_(name.into())
 }
 
 pub fn rel<T: Into<RolePlayerConstraint>>(value: T) -> ThingVariable {
-    UnboundVariable::hidden().rel(value)
+    UnboundConceptVariable::hidden().rel(value)
+}
+
+pub fn eq<T: Into<Value>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Eq, value.into())
+}
+
+pub fn neq<T: Into<Value>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Neq, value.into())
+}
+
+pub fn lt<T: Into<Value>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Lt, value.into())
+}
+
+pub fn lte<T: Into<Value>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Lte, value.into())
 }
 
-pub fn eq<T: Into<Value>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Eq, value.into())
+pub fn gt<T: Into<Value>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Gt, value.into())
 }
 
-pub fn neq<T: Into<Value>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Neq, value.into())
+pub fn gte<T: Into<Value>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Gte, value.into())
 }
 
-pub fn lt<T: Into<Value>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Lt, value.into())
+pub fn contains<T: Into<String>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Contains, Value::from(value.into()))
 }
 
-pub fn lte<T: Into<Value>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Lte, value.into())
+pub fn like<T: Into<String>>(value: T) -> PredicateConstraint {
+    PredicateConstraint::new(token::Predicate::Like, Value::from(value.into()))
 }
 
-pub fn gt<T: Into<Value>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Gt, value.into())
+pub fn abs<T: Into<Expression>>(arg: T) -> Function {
+    Function { function_name: token::Function::Abs, args: vec![Box::from(arg.into())] }
 }
 
-pub fn gte<T: Into<Value>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Gte, value.into())
+pub fn ceil<T: Into<Expression>>(arg: T) -> Function {
+    Function { function_name: token::Function::Ceil, args: vec![Box::from(arg.into())] }
 }
 
-pub fn contains<T: Into<String>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Contains, Value::from(value.into()))
+pub fn floor<T: Into<Expression>>(arg: T) -> Function {
+    Function { function_name: token::Function::Floor, args: vec![Box::from(arg.into())] }
 }
 
-pub fn like<T: Into<String>>(value: T) -> ValueConstraint {
-    ValueConstraint::new(Predicate::Like, Value::from(value.into()))
+pub fn round<T: Into<Expression>>(arg: T) -> Function {
+    Function { function_name: token::Function::Round, args: vec![Box::from(arg.into())] }
 }
```

Comment:
These builders aren't actually visible from outside the crate, you need to export them in `typeql_lang.rs`